thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -44,71 +46,85 @@ import javax.swing.SortOrder
 import javax.swing.UIManager
 import kotlin.io.path.inputStream
 import kotlin.io.path.name
+import kotlin.io.path.nameWithoutExtension
 
 class MultiThreadView(
-    private val paths: List<Path>
+    private val paths: List<Path>,
 ) : ToolPanel() {
-    private val threadDumps = paths.mapNotNull { path -> ThreadDump.fromStream(path.inputStream()) }.toList()
+    private val threadDumps = paths.map { path ->
+        ThreadDump.fromStream(path.inputStream()) ?: throw ToolOpeningException("Failed to open $path as a thread dump")
+    }
+
+    private var visibleThreadDumps: List<ThreadDump> = emptyList()
+        set(value) {
+            field = value
+            currentLifespanList = value.toLifespanList()
+        }
+
+    private var currentLifespanList: List<ThreadLifespan> = emptyList()
+        set(value) {
+            field = value
+            val allThreads = value.flatten().filterNotNull()
+            poolList.model = FilterModel(allThreads.groupingBy(Thread::pool).eachCount())
+            stateList.model = FilterModel(allThreads.groupingBy { it.state.toString() }.eachCount())
+            systemList.model = FilterModel(allThreads.groupingBy(Thread::system).eachCount())
```

Comment:
Directly apply new models so that the _components_ don't need to know about how to get the data in the format they want, they just need to react appropriately to a model update